mesonbuild/modules/rust.py

```diff
@@ -206,7 +206,10 @@ def bindgen(self, state: ModuleState, args: T.List, kwargs: FuncBindgen) -> Modu
             for i in de.get_include_dirs():
                 clang_args.extend([f'-I{x}' for x in i.to_string_list(
                     state.environment.get_source_dir(), state.environment.get_build_dir())])
-            clang_args.extend(de.get_all_compile_args())
+            # replace all -isystem args with -I for now. Fixing this properly is a huge pain as we'd
```

Comment:
Objection! This comment should say "Breaking this properly"! :P

Using -isystem either means that you're actually instrumenting the compiler's logic for finding system headers, or you're breaking the world by adding non-system headers to the compiler's `#include_next` and similar logic in order to trigger ***undocumented*** side effects like ignoring warnings.

Sounds like it's the latter case, and that compiler-mangling is breaking bindgen. I wish people would stop breaking their compiler by using weird options.

Reply:
It comes right from the LLVM cmake stuff

Reply:
Anyway, I think the best would be to do something like in `CLikeCompilerArgs.to_native` but until then you can't use cmake with bindgen as this just fails to compile as the dependencies are processed differently than in C/C++ targets